ui/desktop/tests/acceptance/projects/sessions/session-test.js

```diff
@@ -148,13 +148,126 @@ module('Acceptance | projects | sessions | session', function (hooks) {
     await click(TARGET_CONNECT_BUTTON);
 
     assert.strictEqual(currentURL(), urls.session);
-    assert.dom('credential-panel-header').doesNotExist();
-    assert.dom('credential-panel-body').doesNotExist();
+    assert.dom('.credential-panel-header').doesNotExist();
+    assert.dom('.credential-panel-body').doesNotExist();
     assert
       .dom('[data-test-no-credentials]')
       .hasText(`Connected You can now access ${instances.target.name}`);
   });
 
+  test('visiting session with vault type credentials should display nested data in a key/value format without escape characters', async function (assert) {
+    this.ipcStub.withArgs('cliExists').returns(true);
+    this.ipcStub.withArgs('connect').returns({
+      session_id: instances.session.id,
+      address: 'a_123',
+      port: 'p_123',
+      protocol: 'tcp',
+      credentials: [
+        {
+          credential_source: {
+            id: 'clvlt_4cvscMTl0N',
+            name: 'Credential Library 0',
+            description: 'Source Description',
+            credential_store_id: 'csvlt_Q1HFGt7Jpm',
+            type: 'vault-generic',
+          },
+          secret: {
+            raw: 'eyJhcnJheSI6WyJvbmUiLCJ0d28iLCJ0aHJlZSIsIm9uZSIsInR3byIsInRocmVlIiwib25lIiwidHdvIiwidGhyZWUiLCJvbmUiLCJ0d28iLCJ0aHJlZSJdLCJuZXN0ZWQiOnsiYm9vbCI6dHJ1ZSwibG9uZyI6IjEyMjM1MzQ1NmFzZWRmYTQzd3J0ZjIzNGYyM2FzZGdmYXNkZnJnYXdzZWZhd3NlZnNkZjQiLCJzZWNlcmV0Ijoic28gbmVzdGVkIn0sInRlc3QiOiJwaHJhc2UifQ',
+            decoded: {
+              data: {
+                backslash: 'password\\withslash\\',
+                email: {
+                  address: 'test.com',
+                },
+              },
+              metadata: {
+                created_time: '2024-04-12T18:38:36.226715555Z',
+                custom_metadata: null,
+                deletion_time: '',
+                destroyed: false,
+                version: 8,
+              },
+            },
+          },
+        },
+      ],
+    });
+
+    await visit(urls.target);
+    await click(TARGET_CONNECT_BUTTON);
+    assert.strictEqual(currentURL(), urls.session);
+
+    assert.dom('.credential-panel-header').exists();
+    assert.dom('.credential-panel-body').exists();
+    assert.dom('.credential-secret').exists();
+
+    // Check if nested data is displayed in a key/value format without escape characters
+    assert
+      .dom('.secret-container:nth-of-type(1)')
+      .hasText('backslash ■ ■ ■ ■ ■ ■ ■ ■ ■ ■ ■ ■');
+    await click('.secret-container:nth-of-type(1) .hds-icon');
+    const expectedOutput = String.raw`password\withslash\ `;
```

Comment:
I added a space at the end to avoid linter error since I'm using a backslash and we want to test the UI displays this output with no additional esc chars.

Reply:
Are you relying on `hasText` trimming whitespace? Wouldn't it be different output then?

What's the linter issue, is it because of the trailing backslash which confuses it? If that's the issue, why not have the backslashes just in the middle instead (and maybe include things like `\t`, `\0`, etc)

Reply:
I switched to `strictEqual` to be more accurate, the `assert.Dom` and `hasText` did not throw an error when there was an additional space at the end, I believe it ignores the white space

Reply:
How come the new assert uses a `trim()` then? Does the text normally have trailing or leading whitespace?

[hasText](https://github.com/mainmatter/qunit-dom/blob/master/API.md#hastext) normally strips the whitespace

Reply:
I don't know if it normally has whitespaces, but I added trim() as a precaution in case of whitespaces

Reply:
If that's the case, what's the difference between what you changed it to vs using `hasText`? Since it seems like we're stripping whitespace either way

Reply:
yeah, it's pretty much the same now,  I didn't want to rely on `hasText` to remove whitespaces and wanted to be more readable with trim and strictequal but yeah we could just use hasText too.

Reply:
Yeah I think it'd be better to just stick with `assert.dom` and `hasText` then since it seems they do the same thing